syntax = "proto2";

package crypto.pb;

enum KeyType {
  RSA = 0;
  Ed25519 = 1;
  Secp256k1 = 2;
  ECDSA = 3;
}

message PublicKey {
  required KeyType key_type = 1;
  required bytes data = 2;
}

message PrivateKey {
  required KeyType key_type = 1;
  required bytes data = 2;
}